packages/app/cypress/e2e/runner/sessions.ui.cy.ts

```diff
@@ -315,34 +313,375 @@ describe('runner/cypress sessions.ui.spec', {
+          {
+            testCase: 'rejects with false',
+            systemTestTitle: 'validate - rejects with false',
+            errMessage: 'rejected with false.',
+          },
+          {
+            testCase: 'promise resolved false',
+            systemTestTitle: 'validate - promise resolves false',
+            errMessage: 'promise resolved false.',
+          },
+          {
+            testCase: 'throws an error',
+            systemTestTitle: 'validate - throws an error',
+            errMessage: 'Something went wrong!',
+            hasCodeFrame: false,
+          },
+        ].forEach(({ testCase, systemTestTitle, errMessage, hasCodeFrame }, index) => {
+          it(`has test error when validate ${testCase}`, () => {
+            cy.contains('.test', systemTestTitle).as('example_test')
+
+            cy.get('@example_test').within(() => {
+              assertRecreatedSession({
+                testAlias: '@example_test',
+                validationErrMessage: errMessage,
+                commandPassed: true,
+                successfullyRecreatedSession: true,
+                hasCodeFrame,
+              })
+            })
+
+            cy.get('@example_test')
+            .find('.attempt-error-region')
+            .should('not.exist')
+          })
         })
       })
 
-      // has error
-      cy.get('@setup_failed').contains('This error occurred while creating session. Because the session setup failed, we failed the test.')
+      describe('failed to recreated session', () => {
+        before(() => {
+          loadSpec({
+            projectName: 'session-and-origin-e2e-specs',
+            filePath: 'session/errors.cy.js',
+            passCount: 0,
+            failCount: 7,
+            setup () {
+              cy.window().then((win) => {
+                // @ts-ignore
+                return win.CYPRESS_TEST_DATA = {
+                  restoreSessionWithValidationFailure: true,
+                  successfullyRecreatedSession: false,
+                }
+              })
+            },
+          })
+        })
+
+        it('setup has failing command', () => {
+          cy.contains('.test', 'setup - has failing command').as('example_test')
+
+          cy.get('@example_test').within(() => {
+            assertRecreatedSession({
+              testAlias: '@example_test',
+              validationErrMessage: 'callback yielded false',
+              commandPassed: false,
+              successfullyRecreatedSession: false,
+            })
+          })
+
+          const recreatedErrPostfix = 'This error occurred while creating the session. Because the session setup failed, we failed the test.'
+
+          cy.get('@example_test')
+          .find('.attempt-error-region')
+          .contains('Expected to find element')
+          .contains(recreatedErrPostfix)
+        })
+
+        describe('failed validation', () => {
+          [
+            {
+              testCase: 'has failing Cypress command',
```

Comment:
| Before | After |
| - | - |
| https://user-images.githubusercontent.com/14099737/199744672-16ebfb96-d73e-4a03-b7f9-7217a4996d11.mov| https://user-images.githubusercontent.com/14099737/199744826-2b7aeffb-d37e-4e04-af76-3cf4e01e0fce.mov

Reply:
It doesn't seem correct that `cy.get()` failing is reported as an `AssertionError`. That should be a `CypressError`.

Reply:
👍🏻 Fixed: [cypress-io/cypress@`28f6add` (#24511)](https://github.com/cypress-io/cypress/pull/24511/commits/28f6add42b8146c1fb95ea01f1e48d868d0a5f5e)